pyroll/core/profile/profile.py

```diff
@@ -622,3 +645,86 @@ def __attrs__(self):
             diagonal=self._diagonal,
             corner_radius=self._corner_radius
         )
+
+
+class HexagonProfile(Profile):
+    def __init__(
+            self,
+            side: Optional[float] = None,
+            diagonal: Optional[float] = None,
+            corner_radius: float = 0,
+            **kwargs
+    ):
+        """
+        Creates a hexagonal shaped profile (a real hexagonal shape with rounded corners,
+        without imperfections of hexagonal grooves). A hexagon is oriented to stand on its side.
+        Give exactly one of ``side`` and ``height``.
+
+        :param side: the side length of the hexagonal profile, must be > 0
+        :param diagonal: the height of the hexagonal profile when standing on the flat base, must be > 0
```

Comment:
May be better to define diagonal as the corner-to-corner distance and height as the side-to-side distance distinctly.

Reply:
so you would measure including the radius or where the projected sides would meet?

Reply:
The side intersection, so this is coherent with squares.